### PR TITLE
Close custom URLClassLoader

### LINKED ISSUE #1896: Custom URLClassLoader is not closed
An instance of [URLClassLoader](https://github.com/junit-team/junit5/blob/master/junit-platform-console/src/main/java/org/junit/platform/console/tasks/ConsoleTestExecutor.java#L81) is created for additional class path entries -- but it is never closed.

## Deliverables

- [ ] Close the instance after usage to free file handles gracefully

### PR DESCRIPTION
**Pull request by sormuras**
## Overview

Closes #1896

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [ ] There are no TODOs left in the code
- [ ] Method [preconditions](http://junit.org/junit5/docs/snapshot/api/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [ ] [Coding conventions](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [ ] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [ ] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/master/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [ ] Change is documented in the [User Guide](http://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](http://junit.org/junit5/docs/snapshot/user-guide/#release-notes)
- [ ] All [continuous integration builds](https://github.com/junit-team/junit5#continuous-integration-builds) pass

### CODE REVIEW

File: junit-platform-console/src/main/java/org/junit/platform/console/tasks/CustomContextClassLoaderExecutor.java [hunk 1]
```diff
@@ -10,38 +10,46 @@
...
 			return callable.call();
 		}
 		finally {
 			Thread.currentThread().setContextClassLoader(originalClassLoader);
+			try {
+				customClassLoaderOrNull.close();
+			}
+			catch (Throwable t) {
+				BlacklistedExceptions.rethrowIfBlacklisted(t);
+				// Ignore or log unchecked exceptions and errors?
```
Comment (sbrannen) [inline junit-platform-console/src/main/java/org/junit/platform/console/tasks/CustomContextClassLoaderExecutor.java]: The method is already declared with `throws Exception`. So why not just let it propagate (i.e., remove the try-catch block)?

**Comment by sormuras** [comment]
Superseded by https://github.com/junit-team/junit5/commit/0fbfdffc9ae7a2337b4096da7f95b4af0f9a6d96
